crates/api/src/module.rs

```diff
@@ -88,6 +90,7 @@ struct ModuleInner {
     imports: Box<[ImportType]>,
     exports: Box<[ExportType]>,
     name: Option<String>,
+    compiled: RefCell<Option<CompiledModule>>,
```

Comment:
Could the interior mutability be avoided here and the compilation happen on construction?

(that's what I believe the constructor of the `Module` is supposed to do anyway)

Reply:
Done! While working on this I noticed that there is a little discrepancy. One cannot create a `Module` without going through validation with a supplied name.